peer_heap.js

```diff
@@ -104,13 +97,20 @@ PeerHeap.prototype.choose = function choose(threshold, filter) {
 
         var el = self.array[i];
 
-        if (self.rangehis[i] <= self.maxRangeStart) {
+        if (!el || (
+            self.rangehis[i] <= self.maxRangeStart &&
+            (!filter && !notEnoughPeers)
+        )) {
             // This range ends before the range with the largest start begins,
             // so it can't possibly be chosen over any of the ranges we've
             // seen. All ranges below this one have a smaller end, so this
             // range and any below it can't be chosen.
             continue;
-        } else if (!filter || filter(el.peer)) {
+        }
+
+        if (!filter || filter(el.peer)) {
+            isSecondary = notEnoughPeers && el.peer.isConnected('out');
```

Comment:
yeah i benched it. very little impact. 

because `notEnoughPeers` is false the majority of the time.